app/src/main/java/nerd/tuxmobil/fahrplan/congress/models/ScheduleData.kt

```diff
@@ -9,4 +9,13 @@ data class ScheduleData(
 
     val roomNames: List<String>
         get() = roomDataList.map { it.roomName }
+
+    val allLectures: List<Lecture>
+        get() = roomDataList.flatMap { it.lectures }.sortedBy { it.dateUTC }
+
+    fun findLecture(lectureId: String): Lecture? {
+        return roomDataList.asSequence()
+                .flatMap { it.lectures.asSequence() }
+                .firstOrNull { it.lectureId == lectureId }
```

Comment:
I wonder if a lecture (`it`) can be `null` and could cause a `NullPointerException`. To avoid this you could reverse the comparison as in `lectureId == it.lectureId`.

Reply:
The list is defined in Kotlin code and populated with Kotlin code. It shouldn't be possible for `null` elements to be in there.
Reversing the comparison has no meaningful effect in Kotlin. See https://kotlinlang.org/docs/reference/equality.html#structural-equality

Reply:
Oh :bulb: Thx for the link. I wasn't aware of this.